bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsAccountHandler.java

```diff
@@ -0,0 +1,300 @@
+                    response.forEach(d -> devices.put(d.getDevTid(), d));
+                }
+                this.devices = devices;
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                if (accountException == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            accountException.getLocalizedMessage());
+                }
+            }
+
+            return null;
+        });
+    }
+
+    private void login() {
+        logger.debug("Login");
+        getAccessToken().handle((accessToken, accountException) -> {
+            this.accessToken = accessToken;
+            if (accessToken != null) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                if (accountException == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            accountException.getLocalizedMessage());
+                }
+            }
+
+            return null;
+        });
+    }
+
+    private CompletableFuture<@Nullable String> getAccessToken() {
+        CompletableFuture<@Nullable String> f = new CompletableFuture<>();
+        Request request = client.newRequest(URI.create(ELRO_CLOUD_LOGIN_URL));
+
+        request.method(HttpMethod.POST).content(new StringContentProvider(loginJson), "application/json")
+                .timeout(TIMEOUT_MS, TimeUnit.MILLISECONDS).send(new BufferingResponseListener() {
+                    @NonNullByDefault({})
+                    @Override
+                    public void onComplete(Result result) {
+                        if (result.isSucceeded()) {
+                            final HttpResponse response = (HttpResponse) result.getResponse();
+                            if (response.getStatus() == 200) {
+                                try {
+                                    Map<String, String> content = gson.fromJson(getContentAsString(), loginType);
+                                    String accessToken = (content != null) ? content.get("access_token") : null;
+                                    f.complete(accessToken);
+                                } catch (JsonParseException parseException) {
+                                    logger.error("Parsing failed: {}", parseException.getMessage());
+                                    f.completeExceptionally(
+                                            new ElroConnectsAccountException("@text/offline.login-parsing-fail"));
+                                }
+                            } else {
+                                f.completeExceptionally(new ElroConnectsAccountException(getContentAsString()));
```

Comment:
If I'm reading this correctly, this exception is caught in `login()` and full content is set as `ThingStatusDetail`? This could possibly be quite long and/or technical?

Reply:
I revisited it. I have heard conflicting messages before, sometimes people wanting to have all info in the UI if possible, sometimes favouring detailed info in the logs.
I now just put a general messages in the thing status, and have all the details when appropriate on warn level in the logs (I had some error messages in the log before, which I probably should not do). I hope this makes more sense. And thank you for pointing it out.

Reply:
Looks good. I don't think there are strict rules: https://www.openhab.org/docs/developer/bindings/#updating-the-thing-status. It can be technical - as in your previous implementation, so that part of my comments was misleading. However, my comment/question was mainly if `getContentAsString()` could risk being anything (like a long HTML page), in which case it would probably not be a good idea to directly use this.